src/components/cart-row/cart-row.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState } from "react";
+import { Link } from "react-router-dom";
+import { CartItem } from "../../entities/cart-item";
+import CartService from "../../services/cart-service";
+
+interface Props {
+  cartItem: CartItem;
+  cart: CartItem[];
+  setCart: (newCart: CartItem[]) => void;
+}
+
+const CartRow = ({ cartItem, cart, setCart }: Props) => {
+  const { product, quantity } = cartItem;
+  const [newQuantity, setQuantity] = useState(quantity);
+
+  const handleQuantityChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newQuantity = parseInt(event.target.value, 10);
+    setQuantity(newQuantity);
+    setCart(CartService.updateCartItemQuantity(cartItem, cart, newQuantity));
+  };
+  return (
+    <div className="cart-row">
+      <Link to={`/products/${product.id}`}><img src={product.imageUrl} width="30" height="30" /></Link>
+      <span className="product-name">{product.name}</span>
+      <span className="product-category">{product.category.name}</span>
+      <input
+        type="number"
+        id="product-quantity"
```

Comment:
It's not a really good practice to use ids (https://medium.com/@clairecodes/reasons-not-to-use-ids-in-css-a42204fb0d97) so maybe you can get rid of this one and use a class instead.

Reply:
Ok. 
I thought it is acceptable with respect to input elements inside a form. But I've used it for styling so you're right.